docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,108 @@
+- Generates certificates and kubeconfig files, writes a kubeconfig locally to `test/kubemark/kubeconfig.loc` and creates a Secret which stores kubeconfig for HollowKubelet/HollowProxy use (used gcloud to transfer files to Master, should be easy to do outside of GCE).
+- Creates a ReplicationController for HollowNodes and starts them up. (will work exactly the same everywhere as long as MASTER_IP will be populated correctly, but you’ll need to update docker image address if you’re not using GCR and default image name)
+
+### Running Density tests on Kubemark cluster
+
+To run standard density ‘30 test on your Kubemark cluster created in the previous step you execute `test/kubemark/run-scalability-test.sh` script. It will configure ginkgo to use Kubemark cluster instead of something else and start a density test. This script should not need any changes to work on other cloud providers.
+Monitoring test execution and debugging problems
+Run-scalability-test prints the same output on Kubemark as on ordinary e2e cluster, but if you need to dig deeper you need to learn how to debug HollowNodes and how Master machine (currently) differs from the ordinary one.
+
+If you need to debug master machine you can do similar things as you do on your ordinary master. The difference between Kubemark setup and ordinary setup is that in Kubemark etcd is run as a plain docker container, and all master components are run as normal processes. There’s no Kubelet overseeing them. Logs are stored in exactly the same place, i.e. `/var/logs/` directory.
+
+To help you with debugging from inside the cluster startup script puts a `~/configure-kubectl.sh` script on the master. It downloads `gcloud` and `kubectl` tool and configures kubectl to work on unsecured master port (useful if there are problems with security). After the script is run you can use kubectl command from the master machine to play with the cluster.
+
+Debugging HollowNodes is a bit more tricky, as if you experience a problem on one of them you need to learn which hollow-node pod corresponds to a given HollowNode known by the Master. When registering HollowNodes use their cluster IPs as names, which means that if you need to find a HollowNode named `10.2.4.5` you just need to find a Pod in external cluster with this cluster IP. There’s a helper script `test/kubemark/get-real-pod-for-hollow-node.sh` that does this for you.
+
+When you have a Pod name you can use `kubectl logs` on external cluster to get logs, or use a `kubectl describe pod` call to find an external Node on which this particular HollowNode is running so you can ssh to it.
+
+All those things should work exactly the same on all cloud providers.
+
+### Turning down Kubemark cluster
+
+On GCE you just need to execute `test/kubemark/stop-kubemark.sh` script, which will delete HollowNode ReplicationController and all the resources (except Firewall Rule) for you. On other providers you’ll need to delete all this stuff by yourself.
```

Comment:
Why it doesn't remove firewall rule?
I think that start-kubemark and stop-kubemark should be symetric.